cpp/src/groupby/sort/group_scan.hpp

```diff
@@ -80,9 +80,42 @@ std::unique_ptr<column> max_scan(column_view const& values,
  * @param stream CUDA stream used for device memory operations and kernel launches.
  * @return Column of type INT32 of count values
  */
-std::unique_ptr<column> count_scan(cudf::device_span<size_type const> group_labels,
+std::unique_ptr<column> count_scan(device_span<size_type const> group_labels,
                                    rmm::cuda_stream_view stream,
                                    rmm::mr::device_memory_resource* mr);
+
+/**
+ * @brief Internal API to calculate groupwise rank value
+ *
+ * @param order_by column or struct column that rows within a group are sorted by
+ * @param group_labels ID of group that the corresponding value belongs to
+ * @param group_offsets group index offsets with group ID indices
+ * @param stream CUDA stream used for device memory operations and kernel launches.
+ * @param mr Device memory resource used to allocate the returned column's device memory
+ * @return Column of type size_type of rank values
+ */
+std::unique_ptr<column> rank_scan(column_view const& order_by,
+                                  device_span<size_type const> group_labels,
+                                  device_span<size_type const> group_offsets,
+                                  rmm::cuda_stream_view stream,
+                                  rmm::mr::device_memory_resource* mr);
+
+/**
+ * @brief Internal API to calculate groupwise dense rank value
+ *
+ * @param order_by column or struct column that rows within a group are sorted by
+ * @param group_labels ID of group that the corresponding value belongs to
+ * @param group_offsets group index offsets with group ID indices
+ * @param stream CUDA stream used for device memory operations and kernel launches.
+ * @param mr Device memory resource used to allocate the returned column's device memory
```

Comment:
Same here....
```suggestion
 * @param group_offsets group index offsets with group ID indices
 * @param stream CUDA stream used for device memory operations and kernel launches
 * @param mr Device memory resource used to allocate the returned column's device memory
```

or
```suggestion
...
 * @param group_offsets group index offsets with group ID indices.
 * @param stream CUDA stream used for device memory operations and kernel launches.
 * @param mr Device memory resource used to allocate the returned column's device memory.
 ...
```